# Week 7: Write a reusable alignment Makefile 

## Add additional code to your Makefile to also create bigWig coverage tracks. See WIGGLE: Genome coverage Links to an external site. page for code.

This file is located in the Week-7 directory. 

## In your README.md demonstrate the use of your Makefile to generate a BAM file for both the original data and the second sequencing data obtained with a different instrument.

### Usage
Each steps of this pipeline is defined as a separate **Makefile target**, allowing you to run them individually. More about the usage can be found in the Week-6 README.md but "wiggle" is well detailed here. 

| **Target** | **Description**                                          |
| ---------- | -------------------------------------------------------- |
| `genome`   | Downloads and extracts the Zika virus genome             |
| `reads`    | Downloads reads from SRA                                 |
| `index`    | Indexes the reference genome using `bwa index`           |
| `align`    | Aligns reads, sorts SAM to BAM, and indexes the BAM file |
| `wiggle`   | Covert the BAM file into a BigWig file                 |
| `clean`    | Removes all generated files and folders                  |

Example Usage
### Step-by-Step Usage
### 1. Downloads and extracts the Zika virus genome.
```
make genome GENOME_ACC=GCF_000882815.3
```

### 2. Downloads sequencing reads from the SRA.
``` 
make reads SRR_ACC=SRR1972739
```

### 3. Indexes the reference genome using BWA.
```
make index REF=ref/zika_paper_genome.fa
```

### 4. Aligns reads to the genome, sorts the result, and creates an indexed BAM.
```
make align SRR_ACC=SRR1972739 REF=ref/zika_paper_genome.fa
```

### 5. wiggle: Coverts the BAM file into a BigWig file
```
make wiggle SRR_ACC=SRR1972739 REF=ref/zika_paper_genome.fa
```

### Extra: Clean Up 
``` 
make clean
```
Input:

Sorted BAM file: bam/SRR3194431.bam

Reference genome: refs/zika_paper_genome.fa

Output:

BigWig file: bam/SRR3194431.bw

Note: This step generates a bedGraph using bedtools genomecov, then converts it to BigWig using bedGraphToBigWig.
Requires a FASTA index (.fai) generated by samtools faidx.
make bigwig depends on make align.

## Visualize the GFF annotations and both wiggle and BAM files in IGV all next to one other so that you can visually compare the two alignment datasets.
<img width="1440" height="900" alt="Screenshot 2025-10-13 at 12 06 49â€¯AM" src="https://github.com/user-attachments/assets/2d56a928-4b97-4054-8935-c1560b61d4c8" />

Command for second alignment dataset:
```
make all SRA_ACC=SRR3194430 GENOME_ACC=GCF_000865845.1
```

## Briefly describe the differences between the alignment in both files.
While both samples aligned to differenct places along the genome, the coverage of at the places the samples aligned were very similar. 

## Briefly compare the statistics for the two BAM files.
Command:
```
samtools flagstat SRR3194430.bam
```

Output
```
1500 + 0 in total (QC-passed reads + QC-failed reads)
1500 + 0 primary
0 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
2 + 0 mapped (0.13% : N/A)
2 + 0 primary mapped (0.13% : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```

Command:
```
samtools flagstat SRR3194431.bam
```

Output
```
1500 + 0 in total (QC-passed reads + QC-failed reads)
1500 + 0 primary
0 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
6 + 0 mapped (0.40% : N/A)
6 + 0 primary mapped (0.40% : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```

This may be because only a small percentage mapped to the reference genome, but the samples are very similar. The only difference being that SRR3194431 mapped the the reference genome more. 

## How many primary alignments does each of your BAM files contain?

Command:
```
samtools view -c -F 0x100 SRR3194430.bam
```

Output:
```
1500
```

Command:
```
samtools view -c -F 0x100 SRR3194431.bam
```
Output:
```
1500
```

Both contain 1500 prinary alignments. 

## What coordinate has the largest observed coverage (hint samtools depth)

Command:
```
samtools depth SRR3194430.bam | sort -k3,3nr | head -n 1
```
Output:
```
NC_012532.1     3741    1
```
Command:
```
samtools depth SRR3194431.bam | sort -k3,3nr | head -n 1
```
Output:
```
NC_012532.1     10179   2
```
## Select a gene of interest. How many alignments on a forward strand cover the gene?
Command:
```
grep 'gene' genomic.gff | sed -n 's/.*Name=\([^;]*\).*/\1/p' | sort -u
```

Output:
```
POLY
YP_002790881.1
```
Command:
```
grep -i "YP_002790881.1" genomic.gff 
```

Output:
```
NC_012532.1     RefSeq  CDS     107     10366   .       +       0       ID=cds-YP_002790881.1;Parent=gene-ZIKV_gp1;Dbxref=GenBank:YP_002790881.1,GeneID:7751225;Name=YP_002790881.1;gbkey=CDS;gene=POLY;locus_tag=ZIKV_gp1;product=polyprotein;protein_id=YP_002790881.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    107     472     .       +       .       ID=id-YP_002790881.1:1..122;Parent=cds-YP_002790881.1;gbkey=Prot;product=anchored capsid protein ancC;protein_id=YP_009227206.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    107     418     .       +       .       ID=id-YP_002790881.1:1..104;Parent=cds-YP_002790881.1;gbkey=Prot;product=capsid protein C;protein_id=YP_009227196.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    473     976     .       +       .       ID=id-YP_002790881.1:123..290;Parent=cds-YP_002790881.1;gbkey=Prot;product=membrane glycoprotein precursor prM;protein_id=YP_009227197.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    473     751     .       +       .       ID=id-YP_002790881.1:123..215;Parent=cds-YP_002790881.1;gbkey=Prot;product=protein pr;protein_id=YP_009227207.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    752     976     .       +       .       ID=id-YP_002790881.1:216..290;Parent=cds-YP_002790881.1;gbkey=Prot;product=membrane glycoprotein M;protein_id=YP_009227208.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    977     2476    .       +       .       ID=id-YP_002790881.1:291..790;Parent=cds-YP_002790881.1;gbkey=Prot;product=envelope protein E;protein_id=YP_009227198.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    2477    3532    .       +       .       ID=id-YP_002790881.1:791..1142;Parent=cds-YP_002790881.1;gbkey=Prot;product=nonstructural protein NS1;protein_id=YP_009227199.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    3533    4210    .       +       .       ID=id-YP_002790881.1:1143..1368;Parent=cds-YP_002790881.1;gbkey=Prot;product=nonstructural protein NS2A;protein_id=YP_009227200.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    4211    4600    .       +       .       ID=id-YP_002790881.1:1369..1498;Parent=cds-YP_002790881.1;gbkey=Prot;product=nonstructural protein NS2B;protein_id=YP_009227201.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    4601    6451    .       +       .       ID=id-YP_002790881.1:1499..2115;Parent=cds-YP_002790881.1;gbkey=Prot;product=nonstructural protein NS3;protein_id=YP_009227202.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    6452    6832    .       +       .       ID=id-YP_002790881.1:2116..2242;Parent=cds-YP_002790881.1;gbkey=Prot;product=nonstructural protein NS4A;protein_id=YP_009227203.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    6833    6901    .       +       .       ID=id-YP_002790881.1:2243..2265;Parent=cds-YP_002790881.1;gbkey=Prot;product=protein 2K;protein_id=YP_009227209.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    6902    7654    .       +       .       ID=id-YP_002790881.1:2266..2516;Parent=cds-YP_002790881.1;gbkey=Prot;product=nonstructural protein NS4B;protein_id=YP_009227204.1
NC_012532.1     RefSeq  mature_protein_region_of_CDS    7655    10363   .       +       .       ID=id-YP_002790881.1:2517..3419;Parent=cds-YP_002790881.1;gbkey=Prot;product=RNA-dependent RNA polymerase NS5;protein_id=YP_009227205.1
```

Zika virus's genome only code one protein (RDRP), called YP_002790881.1 according to the .gff file. Because of this, all of the alignment cover a part of this gene. 
